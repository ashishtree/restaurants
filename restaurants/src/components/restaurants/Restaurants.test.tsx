import * as React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import RestaurantComponent from "./Restaurants";
import RestaurantMockData from "../../__mockData__/mockRestaurants.json";
import * as ShallowRenderer from "react-test-renderer/shallow";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    getRestaurantsData: jest.fn(),
    searchRestaurants: jest.fn(),
    status: "LOADING",
    RestaurantsData: RestaurantMockData,
  };
  const mockStore = configureStore();
  const store = mockStore({});

  const myShallowRenderer = ShallowRenderer.createRenderer();

  const wrapper = myShallowRenderer.render(
    <Provider store={store}>
      <RestaurantComponent {...props} />
    </Provider>
  );

  return {
    props,
    wrapper,
  };
}

describe("Restaurant Component Render", () => {
  describe("Restaurant Component Render", () => {
    it("should render Restaurant Component", () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
