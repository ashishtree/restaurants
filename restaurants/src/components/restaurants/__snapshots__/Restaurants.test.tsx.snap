// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant Component Render Restaurant Component Render should render Restaurant Component 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Memo(Connect(RestaurantComponent))
    RestaurantsData={
      Array [
        Object {
          "favorite": false,
          "id": 1,
          "image": "http://localhost:3006/images/restaurant_logo.jpg",
          "name": "Tanoshii Sushi",
          "sortingValues": Object {
            "averageProductPrice": 1536,
            "bestMatch": 0,
            "deliveryCosts": 200,
            "distance": 1190,
            "minCost": 1000,
            "newest": 96,
            "popularity": 17,
            "ratingAverage": 4.5,
            "shortStatusName": "oa",
            "topRestaurant": 20234.5,
          },
          "status": "open",
        },
      ]
    }
    getRestaurantsData={[MockFunction]}
    searchRestaurants={[MockFunction]}
    status="LOADING"
  />
</Context.Provider>
`;
